packages/eslint-plugin/src/rules/keyword-spacing.ts

```diff
@@ -0,0 +1,598 @@
+     * around those keywords is invalid.
+     * @param node A node to report.
+     */
+    function checkSpacingForForInStatement(node: TSESTree.ForInStatement) {
+      checkSpacingAroundFirstToken(node);
+      checkSpacingAroundTokenBefore(node.right);
+    }
+
+    /**
+     * Reports `for` and `of` keywords of a given node if usage of spacing
+     * around those keywords is invalid.
+     * @param node A node to report.
+     */
+    function checkSpacingForForOfStatement(node: TSESTree.ForOfStatement) {
+      if (node.await) {
+        checkSpacingBefore(sourceCode.getFirstToken(node, 0)!);
+        checkSpacingAfter(sourceCode.getFirstToken(node, 1)!);
+      } else {
+        checkSpacingAroundFirstToken(node);
+      }
+      checkSpacingAround(
+        sourceCode.getTokenBefore(node.right, isNotOpeningParenToken)!,
+      );
+    }
+
+    /**
+     * Reports `import`, `export`, `as`, and `from` keywords of a given node if
+     * usage of spacing around those keywords is invalid.
+     *
+     * This rule handles the `*` token in module declarations.
+     *
+     *     import*as A from "./a"; /*error Expected space(s) after "import".
+     *                               error Expected space(s) before "as".
+     * @param node A node to report.
+     */
+    function checkSpacingForModuleDeclaration(
+      node:
+        | TSESTree.ExportNamedDeclaration
+        | TSESTree.ExportDefaultDeclaration
+        | TSESTree.ExportAllDeclaration
+        | TSESTree.ImportDeclaration,
+    ) {
+      const firstToken = sourceCode.getFirstToken(node)!;
+
+      checkSpacingBefore(firstToken, PREV_TOKEN_M);
+      checkSpacingAfter(firstToken, NEXT_TOKEN_M);
+
+      if (node.type === 'ExportDefaultDeclaration') {
+        checkSpacingAround(sourceCode.getTokenAfter(firstToken)!);
+      }
+
+      if ((node as any).source) {
+        const fromToken = sourceCode.getTokenBefore((node as any).source)!;
+
+        checkSpacingBefore(fromToken, PREV_TOKEN_M);
+        checkSpacingAfter(fromToken, NEXT_TOKEN_M);
+      }
+    }
+
+    /**
+     * Reports `as` keyword of a given node if usage of spacing around this
+     * keyword is invalid.
+     * @param node A node to report.
+     */
+    function checkSpacingForImportNamespaceSpecifier(node: TSESTree.Node) {
+      const asToken = sourceCode.getFirstToken(node, 1)!;
+
+      checkSpacingBefore(asToken, PREV_TOKEN_M);
+    }
+    /**
+     * Reports `static`, `get`, and `set` keywords of a given node if usage of
+     * spacing around those keywords is invalid.
+     * @param node A node to report.
+     */
+    function checkSpacingForProperty(
+      node: TSESTree.MethodDefinition | TSESTree.Property,
+    ) {
+      if ((node as any).static) {
+        checkSpacingAroundFirstToken(node);
+      }
+      if (
+        node.kind === 'get' ||
+        node.kind === 'set' ||
+        (((node as any).method || node.type === 'MethodDefinition') &&
```

Comment:
Any way to avoid these ugly `as any`?

Reply:
You can do checks against the [discriminatory property](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions) to narrow the types, which will let you access properties without casting :)

For example, in your first `if` condition, you can do:

```
if (
  node.type === AST_NODE_TYPES.MethodDefinition &&
  node.static // 'node' is now type TSESTree.MethodDefintion until the closing } or another guard narrows it
) {
  //
}
```

Generally, it's always better to cast to the exact type you need rather than `any` - and from there you can work backwards to figure out what checks you need to do to narrow the type down without the cast.

When working with AST nodes, typically it's about checking their `type`, but with other less uniform objects you might need to write a custom type guard.

Reply:
Got it. Although the problem doesn't exist anymore since I import the base rule.